src/Features/LanguageServer/Protocol/Handler/Diagnostics/DiagnosticSources/DocumentDiagnosticSource.cs

```diff
@@ -9,19 +9,17 @@
 
 namespace Microsoft.CodeAnalysis.LanguageServer.Handler.Diagnostics;
 
-internal sealed record DocumentDiagnosticSource(Document Document) : AbstractDocumentDiagnosticSource<Document>(Document)
+internal sealed record DocumentDiagnosticSource(DiagnosticKind DiagnosticKind, Document Document)
+    : AbstractDocumentDiagnosticSource<Document>(Document)
 {
-    // The normal diagnostic source includes both todo comments and diagnostics for this open file.
-    protected override bool IncludeTaskListItems => true;
-    protected override bool IncludeStandardDiagnostics => true;
-
-    protected override async Task<ImmutableArray<DiagnosticData>> GetDiagnosticsWorkerAsync(
+    public override async Task<ImmutableArray<DiagnosticData>> GetDiagnosticsAsync(
         IDiagnosticAnalyzerService diagnosticAnalyzerService, RequestContext context, CancellationToken cancellationToken)
     {
         // We call GetDiagnosticsForSpanAsync here instead of GetDiagnosticsForIdsAsync as it has faster perf
         // characteristics. GetDiagnosticsForIdsAsync runs analyzers against the entire compilation whereas
         // GetDiagnosticsForSpanAsync will only run analyzers against the request document.
-        var allSpanDiagnostics = await diagnosticAnalyzerService.GetDiagnosticsForSpanAsync(Document, range: null, cancellationToken: cancellationToken).ConfigureAwait(false);
+        var allSpanDiagnostics = await diagnosticAnalyzerService.GetDiagnosticsForSpanAsync(
+            Document, range: null, diagnosticKind: this.DiagnosticKind, cancellationToken: cancellationToken).ConfigureAwait(false);
         return allSpanDiagnostics;
     }
```

Comment:
this part was extremely clean, and fits onto the current diagnostic service API without any issue.